src/cljdoc/storage/postgres_impl.clj

```diff
@@ -0,0 +1,46 @@
+(ns cljdoc.storage.postgres-impl
+  "Postgres implementation of the abstract storage interface defined in [[cljdoc.storage]].
+  This namespace only contains the Postgres specific parts of the implementation and requires the
+  DB-agnostic parts from `cljdoc.storage.db-commons`."
+  (:require [cljdoc.storage.api :refer [IStorage]]
+            [cljdoc.storage.db-commons :as db]
+            [cljdoc-shared.spec.analyzer :as analyzer-spec]
+            [clojure.java.jdbc :as sql]))
+
+(defn- sql-exists?
+  "A small helper to deal nested Postgres returns for exists queries."
+  [db-spec sqlvec]
+  (val (first (first (sql/query db-spec sqlvec)))))
+
+(defn store-artifact! [db-spec group-id artifact-id versions]
+  (assert (coll? versions))
+  (doseq [v versions]
+    (sql/execute! db-spec ["INSERT INTO versions (group_id, artifact_id, name) VALUES (?, ?, ?) ON CONFLICT DO NOTHING" group-id artifact-id v])))
+
+(defrecord PostgresStorage [db-spec]
```

Comment:
Yes right, I tried to pull as much as possible to the `db-commons` ns to avoid duplication and ended up with these two functions that I could not "generify" further. I'm not really happy with this state yet, do you maybe have some suggestions for improvement here?